controller-server/src/main/java/com/yahoo/vespa/hosted/controller/metric/ConfigServerMetrics.java

```diff
@@ -62,4 +71,20 @@ private double weightedAverageLatency(List<ClusterMetrics> metrics,
         return weightedLatency / rateSum;
     }
 
+    public JsonResponse buildResponseFromProtonMetrics(List<ProtonMetrics> protonMetrics) {
```

Comment:
It's not really this class' responsibility to create HTTP responses, it does metric aggregation.

You can build the response directly in ApplicationApiHandler

Reply:
Done